src/components/forms/Textarea/Textarea.tsx

```diff
@@ -0,0 +1,158 @@
+   */
+  isDisabled?: boolean;
+  /**
+   * Additional description displayed close to the field - use this to document any
+   * validation requirements that are crucial to successfully submit the form. More
+   * information that is contextual/background typically belongs in a tooltip.
+   */
+  description?: React.ReactNode;
+  /**
+   * Optional tooltip to provide additional information that is not crucial but may
+   * assist users in filling out the field correctly.
+   */
+  tooltip?: React.ReactNode;
+  /**
+   * Placeholder when no (default) value is available.
+   */
+  placeholder?: string;
+  /**
+   * Automatic expanding and shrinking of fields to fit their content.
+   */
+  autoExpand?: boolean;
+}
+
+const Textarea: React.FC<TextareaProps & UtrechtTextareaProps> = ({
+  name,
+  label = '',
+  isRequired = false,
+  description = '',
+  isDisabled = false,
+  autoExpand = false,
+  placeholder,
+  tooltip,
+  ...extraProps
+}) => {
+  const {validateField} = useFormikContext();
+  const [{value, ...props}, {error = '', touched}] = useField<string | undefined>({
+    name,
+    type: 'text',
+  });
+  const id = useId();
+  const textareaRef = useRef<HTMLTextAreaElement | null>(null);
+
+  const invalid = touched && !!error;
+  const errorMessageId = invalid ? `${id}-error-message` : undefined;
+
+  // Used in conjunction with `autoExpand` to calculate the textarea height
+  const numberOfLineBreaks = ((value || '').match(/\n/g) || []).length;
+
+  useLayoutEffect(() => {
+    const textarea = textareaRef.current;
+    if (!textarea || !autoExpand) {
+      return;
+    }
+
+    const calculateTextareaHeight = (rows: number): number => {
+      // Style values are returned as px values.
+      const {lineHeight, paddingBlockStart, paddingBlockEnd, borderBlockStart, borderBlockEnd} =
+        getComputedStyle(textarea);
+
+      // Parse all style values to float, to remove the trailing "px", and make them calculable.
+      return (
+        rows * parseFloat(lineHeight) +
+        parseFloat(paddingBlockStart) +
+        parseFloat(paddingBlockEnd) +
+        parseFloat(borderBlockStart) +
+        parseFloat(borderBlockEnd)
+      );
+    };
+
+    const resizeTextarea = () => {
+      // Include the line without linebreak in the calculation
+      const numberOfLines = numberOfLineBreaks + 1;
+
+      const minimalHeight: number = calculateTextareaHeight(textarea.rows);
+      const requiredHeight: number = calculateTextareaHeight(numberOfLines);
+
+      const newHeight = Math.max(minimalHeight, requiredHeight);
+      // Only update if new height is different from the current height
+      if (parseFloat(getComputedStyle(textarea).height) !== newHeight) {
+        textarea.style.height = `${newHeight}px`;
+      }
+    };
+
+    resizeTextarea();
+  }, [numberOfLineBreaks, autoExpand]);
+
+  return (
+    <FormField type="text" invalid={invalid} className="utrecht-form-field--openforms">
```

Comment:
```suggestion
    <FormField type="textarea" invalid={invalid} className="utrecht-form-field--openforms">
```
I think it controls the modifier